imports/ui/unit-explorer/unit-explorer.jsx

```diff
@@ -12,62 +12,159 @@ import { setDrawerState } from '../general-actions'
+import { Tabs, Tab } from 'material-ui/Tabs'
+import SwipeableViews from 'react-swipeable-views'
+
+class UnitsByRoles extends Component {
+  render () {
+    const { unitsByRoles, showAddBtn, handleUnitClicked, handleAddCaseClicked, administrate } = this.props
+    return (
+      <div>
+        {unitsByRoles.length === 0 ? (undefined) : (
+          <div className='bb b--black-10 bg-very-light-gray f6 fw5 mid-gray pv2 pl2'>
+            { administrate ? ('Units I Administrate') : ('Units Involved In') }
+          </div>
+        )
+        }
+        {
+          unitsByRoles.map(({ id, name, description, metaData }) => (
+            <MenuItem key={id} innerDivStyle={{padding: 0}} onClick={() => handleUnitClicked(id)} >
+              <div className='mt2 ph2 bb b--very-light-gray br1 w-100 flex items-center pa2'>
+                <FontIcon className='material-icons' color='var(--semi-dark-gray)'>home</FontIcon>
+                <div className='ml3 mv1 semi-dark-gray lh-copy flex-grow overflow-hidden'>
+                  <div className='ti1 ellipsis'>{metaData.displayName || name}</div>
+                  <div className='ti1 ellipsis silver'>{ metaData.moreInfo || description}&nbsp;</div>
+                </div>
+                { !showAddBtn ? (
+                  <div
+                    onClick={evt => handleAddCaseClicked(evt, id)}
+                    className='f6 ellipsis ml3 pl1 mv1 bondi-blue fw5 no-shrink'
+                  >
+                    Add case
+                  </div>
+                ) : (undefined)
+                }
+              </div>
+            </MenuItem>
+          ))}
+      </div>
+    )
+  }
+}
+
+class FilteredUnitsList extends Component {
+  render () {
+    const { filteredUnits, currentUserId, active } = this.props
+    const administratedUnits = filteredUnits.filter(unit => unit.metaData.ownerIds && unit.metaData.ownerIds.includes(currentUserId))
+    const unitsInvolvedIn = filteredUnits.filter(unit => !unit.metaData.ownerIds || !unit.metaData.ownerIds.includes(currentUserId))
+    const divProps = Object.assign({}, this.props)
+    delete divProps.active
+    delete divProps.currentUserId
+
+    return (
+      <div className='flex flex-column'>
+        {filteredUnits.length === 0 ? (
+          <div className='f6 i silver ba b--moon-gray mt2 pa2 tc br1 mr3 ml3'>
+            { !active ? ('You have no units managed with Unee-T yet') : ('You have no disabled units') }
+          </div>
+        ) : <div>
+          <UnitsByRoles {...divProps} unitsByRoles={administratedUnits} administrate />
+          <UnitsByRoles {...divProps} unitsByRoles={unitsInvolvedIn} />
+        </div>
+        }
+      </div>
+    )
+  }
+}
 
 class UnitExplorer extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      slideIndex: 0
+    }
+  }
+
+  handleChange = (value) => {
+    this.setState({
+      slideIndex: value
+    })
+  };
+
+  handleAddCaseClicked (evt, id) {
```

Comment:
I know that in some components the handler's definition is similar to this, and in others it is similar to the way you have "handleChange" defined. The way "handleChange" is defined is actually better as it doesn't require `.bind(this)` to be done on the handler when it is passed down to props. Anyway, it should be consistent across a single component and not use both definition types within a single one. Please change "handleAddCaseClicked" and "handleUnitClicked" to be defined like "handleChange"